physicalplan/scan/vector_selector.go

```diff
@@ -31,27 +31,29 @@ type vectorSelector struct {
 	once       sync.Once
 	vectorPool *model.VectorPool
 
-	mint        int64
-	maxt        int64
-	step        int64
-	currentStep int64
-	stepsBatch  int
+	mint          int64
+	maxt          int64
+	step          int64
+	currentStep   int64
+	stepsBatch    int
+	lookbackDelta int64
 
 	shard     int
 	numShards int
 }
 
-func NewVectorSelector(pool *model.VectorPool, storage *seriesSelector, mint, maxt time.Time, step time.Duration, stepsBatch, shard, numShards int) model.VectorOperator {
+func NewVectorSelector(pool *model.VectorPool, storage *seriesSelector, mint, maxt time.Time, step, lookbackDelta time.Duration, stepsBatch, shard, numShards int) model.VectorOperator {
 	// TODO(fpetkovski): Add offset parameter.
 	return &vectorSelector{
 		storage:    storage,
 		vectorPool: pool,
 
-		mint:        mint.UnixMilli(),
-		maxt:        maxt.UnixMilli(),
-		step:        step.Milliseconds(),
-		currentStep: mint.UnixMilli(),
-		stepsBatch:  stepsBatch,
+		mint:          mint.UnixMilli(),
+		maxt:          maxt.UnixMilli(),
+		step:          step.Milliseconds(),
+		currentStep:   mint.UnixMilli(),
+		stepsBatch:    stepsBatch,
+		lookbackDelta: lookbackDelta.Milliseconds(),
```

Comment:
Do we need to add this to the matrix selector too?

Reply:
I am not sure tbh. Need to check promql engine code.
But right now when we selecting points for matrix selector, we don't handle lookback delta now. https://github.com/thanos-community/promql-engine/blob/main/physicalplan/scan/matrix_selector.go#L191

Reply:
Yeah I think the matrix selector should be okay as it is, because it has its own lookback in the form of a select range.